apps/aecore/src/aec_db.erl

```diff
@@ -244,13 +246,29 @@ backend_mode(<<"mnesia">> , #{persist := true } = M) -> M#{module => mnesia,
 transaction(Fun) when is_function(Fun, 0) ->
     mnesia:activity(transaction, Fun).
 
+dirty(Fun) when is_function(Fun, 0) ->
+    mnesia:activity(async_dirty, Fun).
+
 ensure_transaction(Fun) when is_function(Fun, 0) ->
     %% TODO: actually, some non-transactions also have an activity state
-    case get(mnesia_activity_state) of undefined ->
+    case get(mnesia_activity_state) of
+        undefined ->
+            transaction(Fun);
+        {_, _, non_transaction} ->
+            %% Transaction inside a dirty context; rely on mnesia to handle it
```

Comment:
It may be worth to rebase on master, to see more clearly changes of this PR.

Also the definition of macro `t` in this PR is misleading, and different of what will be after merge to master.

Reply:
Actually, I believe that this macro definition is the right one, and the one currently on master was a mistake.

Reply:
This shall be now addressed in 6ca8c386d